tidy3d/plugins/mode/mode_solver.py

```diff
@@ -116,6 +151,7 @@ def solve(self) -> ModeSolverData:
         """
         return self.data
 
+    # pylint: disable=too-many-locals
```

Comment:
I think it could improve readability to break this method up. or at least the group index portion of it. is it possible? Then I took a look at the full `data_raw` and noticed that it's already very long and hard to follow. @momchil-flex maybe this is pretty low priority but the mode solver code probably could be cleaned up a bit in the future. @lucas-flexcompute I'll leave it to you to decide if it's worth it, but I do think throwing a separate method to compute group index could really improve readability

```py
def data_raw(self) -> ModeSolverData:
    if self.compute_group_index > 0:
        return self._compute_group_index()
    ...
```

Reply:
Sure thing, I've separated the group index calculation as you suggested.